File: server/tests/fixtures/alpha_rule_triggers/python/deadcode_unused_public.py
# deadcode.unused_public fixture
# This file defines public symbols that are never used externally
# Should trigger: deadcode.unused_public

class UnusedPublicClass:
    """A public class that is never imported or used."""
    
    def unused_method(self):
        """An unused public method."""
        return "never called"

def unused_public_function():
    """A public function that is never imported anywhere."""
    return "dead code"

PUBLIC_UNUSED_CONSTANT = "never referenced"

# This is internal and is fine
def _private_helper():
    """Private helper - OK to be unused."""
    return "internal"

File: server/tests/fixtures/alpha_rule_triggers/typescript/deadcode_unused_public.ts
// deadcode.unused_public fixture
// This file defines public symbols that are never used externally
// Should trigger: deadcode.unused_public

export class UnusedPublicClass {
    unusedMethod(): string {
        return "never called";
    }
}

export function unusedPublicFunction(): string {
    return "dead code";
}

export const PUBLIC_UNUSED_CONSTANT: string = "never referenced";

// This is internal and is fine
function privateHelper(): string {
    return "internal";
}
